### PR TITLE
wip

### PR DESCRIPTION
**Pull request by PatrickAlphaC**

### CODE REVIEW

**Comment by PatrickAlphaC** [comment]
This is to address:
https://github.com/smartcontractkit/chainlink-hardhat-box/issues/10
https://github.com/smartcontractkit/chainlink-hardhat-box/issues/9
https://github.com/smartcontractkit/chainlink-hardhat-box/issues/8
https://github.com/smartcontractkit/chainlink-hardhat-box/issues/7

File: deploy/00_Deploy_Mocks.js [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+        await deploy('MockOracle', {
+            from: deployer,
+            log: true,
+            args: [linkToken.address]
+        })
         log("Mocks Deployed!")
         log("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
         log("You are deploying to a local network, you'll need a local network running to interact")
-        log("Please run `npx hardhat node` to interact with the deployed smart contracts!")
+        log("Please run `npx hardhat console` to interact with the deployed smart contracts!")
```
Comment (PatrickAlphaC) [inline deploy/00_Deploy_Mocks.js:34]: We should have both `npx hardhat console` and `npx hardhat node` here?

File: deploy/00_Deploy_Mocks.js [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
     // If we are on a local development network, we need to deploy mocks!
     if (chainId == 31337) {
-        log("Local network deteteched! Deploying mocks...")
-        linkToken = await deploy('LinkToken', { from: deployer })
-        await deploy('MockV3Aggregator', { from: deployer, args: [DECIMALS, INITIAL_PRICE] })
-        await deploy('VRFCoordinatorMock', { from: deployer, args: [linkToken.address] })
-        await deploy('MockOracle', { from: deployer, args: [linkToken.address] })
+        log("Local network detected! Deploying mocks...")
+        const linkToken = await deploy('LinkToken', { from: deployer, log: true })
+        await deploy('EthUsdAggregator', {
```
Comment (PatrickAlphaC) [inline deploy/00_Deploy_Mocks.js:15]: Is this confusing? We then refer it back to mockaggregator later?

File: deploy/01_Deploy_PriceConsumerV3.js [hunk 3]
```diff
@@ -7,26 +7,23 @@ module.exports = async ({
 }) => {
     const { deploy, log } = deployments
     const { deployer } = await getNamedAccounts()
-    let chainId = await getChainId()
+    const chainId = await getChainId()
     let ethUsdPriceFeedAddress
     if (chainId == 31337) {
-        MockV3Aggregator = await deployments.get('MockV3Aggregator')
+        const MockV3Aggregator = await deployments.get('EthUsdAggregator')
```
Comment (PatrickAlphaC) [inline deploy/01_Deploy_PriceConsumerV3.js:13]: Is this confusing?

File: deploy/02_Deploy_APIConsumer.js [hunk 4]
```diff
@@ -21,22 +21,19 @@ module.exports = async ({
     linkTokenAddress = networkConfig[chainId]['linkToken']
     oracle = networkConfig[chainId]['oracle']
   }
-  let jobId = networkConfig[chainId]['jobId']
-  let fee = networkConfig[chainId]['fee']
+  const jobId = networkConfig[chainId]['jobId']
+  const fee = networkConfig[chainId]['fee']
 
-  log("----------------------------------------------------")
-  log('Deploying APIConsumer')
```
Comment (PatrickAlphaC) [inline deploy/02_Deploy_APIConsumer.js]: we don't like these?

File: deploy/00_Deploy_Mocks.js [hunk 5]
```diff
@@ -1,27 +1,38 @@
...
+        })
         log("Mocks Deployed!")
         log("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
         log("You are deploying to a local network, you'll need a local network running to interact")
-        log("Please run `npx hardhat node` to interact with the deployed smart contracts!")
+        log("Please run `npx hardhat console` to interact with the deployed smart contracts!")
         log("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
     }
 }
-module.exports.tags = ['Mocks']
```
Comment (PatrickAlphaC) [inline deploy/00_Deploy_Mocks.js]: I don't understand the different tags?

File: test/APIConsumer_test.js [hunk 6]
```diff
@@ -1,36 +1,29 @@
...
   beforeEach(async () => {
-    // This runs the deploy script, and deploys everything in deploy!
-    // We have it setup so that we don't deploy mocks
-    // We ONLY deploy the fixtures on a localnet
-    // If we want to deploy fixtures to a testnet, just run `hh deploy --network network`
-    let chainId = await getChainId()
-    if (chainId == 31337) {
-      await deployments.fixture()
-    }
+    await deployments.fixture('api')
```
Comment (PatrickAlphaC) [inline test/APIConsumer_test.js:7]: won't this also deploy mocks? We don't always want to do that right?

File: deploy/00_Deploy_Mocks.js (hunk 1 above)
Comment (thodges-gh) [reply deploy/00_Deploy_Mocks.js:34]: They would have to run `hardhat node` to run the deployments on their local chain, then `hardhat console` should allow them to interact with the deployed contracts.

File: deploy/00_Deploy_Mocks.js (hunk 2 above)
Comment (thodges-gh) [reply deploy/00_Deploy_Mocks.js:15]: This is to future-proof the deployment. When you can have multiple uses of the same contract, it's best to give them a unique name. In this case `EthUsdAggregator` is specific to the data used by the contract. That way if we wanted to later add a mock for `BtcUsdAggregator` is simple to add it in and use that in a test.

File: deploy/02_Deploy_APIConsumer.js (hunk 4 above)
Comment (thodges-gh) [reply deploy/02_Deploy_APIConsumer.js]: They're already logged with `log: true` in `deploy`.

File: deploy/00_Deploy_Mocks.js (hunk 5 above)
Comment (thodges-gh) [reply deploy/00_Deploy_Mocks.js]: Now you can specify which use-case you want to deploy:
- `all`: Deploy all contracts
- `feed`: Deploy mocks and PriceConsumerV3
- `api`: Deploy mocks and APIConsumer
- `vrf`: Deploy mocks and RandomNumberConsumer

File: test/APIConsumer_test.js (hunk 6 above)
Comment (thodges-gh) [reply test/APIConsumer_test.js:7]: In tests we do.

File: deploy/01_Deploy_PriceConsumerV3.js (hunk 3 above)
Comment (thodges-gh) [reply deploy/01_Deploy_PriceConsumerV3.js:13]: We should actually rename `const MockV3Aggregator` to `const EthUsdAggregator`. I just missed this and will update.

**Pull request merged**